src/wasm-interpreter.h

```diff
@@ -491,6 +491,22 @@ class ExpressionRunner : public Visitor<SubType, Flow> {
   }
 };
 
+// Execute an expression in global init
+class GlobalInitRunner : public ExpressionRunner<GlobalInitRunner> {
+public:
+  Flow visitLoop(Loop* curr) { WASM_UNREACHABLE(); }
```

Comment:
Look at https://github.com/WebAssembly/binaryen/pull/650/files/0dc286f78cd3d5d1f5fd1cb21c16485cd9f2fd1f#diff-33a3856e71772f6165f55c85cb076496R264 , it's actually a stronger check than needed - it only accepts `Const` (because that's all that is tested currently, and I'm not sure what else will be allowed).